libcaf_core/test/detail/rfc3629.cpp

```diff
@@ -2,15 +2,16 @@
 // the main distribution directory for license terms and copyright or visit
 // https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
 
-#define CAF_SUITE detail.rfc3629
-
 #include "caf/detail/rfc3629.hpp"
 
-#include "core-test.hpp"
+#include "caf/test/test.hpp"
 
 using namespace caf;
+using detail::rfc3629;
+
+using res_t = std::pair<size_t, bool>;
```

Comment:
Now that we're on the new testing framework, what do you think of structuring this a bit more? Something like this:

```
TEST("rfc3629::valid checks whether an input is valid UTF-8") {
  SECTION("ASCII input") {
    ...
  }
  ...
}

TEST("rfc3629::validate stops at the first invalid byte") {
  SECTION("....") {
    ...
  }
}

TEST("rfc3629::validate stops at incomplete UTF-8 code points") {
  ...
}
```